### ISSUE TITLE
File watchers doesn't detect files added by rclone mount on Windows

### ISSUE THREAD

**Issue by felipecrs** [rclone/rclone#7985]
#### The associated forum post URL from `https://forum.rclone.org`

Since this is much dev stuff, I thought a post would not be necessary.

#### What is the problem you are having with rclone?

I am trying to watch for new files in a directory mounted with rclone but it doesn't work consistently.

```
node -e "require('chokidar').watch('C:\\Users\\felip\\clipboard-sync').on('all', (event, path) => { console.log(event, path); });"
```

Check this:

https://github.com/user-attachments/assets/805cf6d2-e43b-4a59-957f-099a96160534

And this feature is needed by my application. For example, the same problem doesn't occur when using Windows' native "remote" file system attributes, like the official OneDrive client and Proton Drive uses.

I already tested `--network-mode` with the same result, and several other flags combinations that I thought would make sense.

Any help is appreciated. I could not find anything similar in the internet.

#### What is your rclone version (output from `rclone version`)

```
rclone v1.67.0
- os/version: Microsoft Windows 11 Pro 23H2 (64 bit)
- os/kernel: 10.0.22631.3958 (x86_64)
- os/type: windows
- os/arch: amd64
- go/version: go1.22.4
- go/linking: static
- go/tags: cmount
```

#### Which OS you are using and how many bits (e.g. Windows 7, 64 bit)

Windows 11 64 bit

#### Which cloud storage system are you using? (e.g. Google Drive)

Microsoft OneDrive

#### The command you were trying to run (e.g. `rclone copy /tmp remote:tmp`)

```
rclone mount onedrive-inatel:/clipboard-sync C:\Users\felip\clipboard-sync --vfs-cache-mode full --poll-interval 1s --dir-cache-time 2s
```

#### A log from the command with the `-vv` flag (e.g. output from `rclone -vv copy /tmp remote:tmp`)

Everything works normally, no errors are shown. I don't think there's anything useful in the logs.

#### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

**Comment by ncw** [comment]
Can you research whether winfsp supports this or not? I suspect that it is entirely winfsp's responsibility.

**Comment by felipecrs** [comment]
@ncw, I will try. However, I suspect this happens because the use of sparse files, whose contents are only fully downloaded when some application attempts to read it.

Do you know if there's a way to make rclone not actually use sparse files and indeed download the files on rclone mount?

**Comment by ncw** [comment]
I don't think the sparse files will have anything to do with it. That's invisible to the filling system that things presents.

You could try with --vfs-cache-mode off if you want to be sure

**Comment by felipecrs** [comment]
Right. Indeed it makes no difference (I had tried --vfs-cache-mode off already).

**Comment by felipecrs** [comment]
Indeed the same issue happens on a directory mounted with WinFSP without rclone.

**Comment by felipecrs** [comment]
Just FYI @ncw, I've implemented a workaround in my application:

- https://github.com/felipecrs/clipboard-sync/pull/102

Basically, I added a watch mode that calls `fs.readDir` to list all files in a directory periodically. It's a bit expensive and inefficient, but it works well for now.

I don't know if you advanced in implementing the WinFSP feature already, but I just wanted to let you know that this isn't so important for me.

**Comment by felipecrs** [comment]
I reopened the ticket given that some implementation on rclone side will be required:

- https://github.com/winfsp/winfsp/issues/571#issuecomment-2296101984

**Comment by ncw** [comment]
So we've worked out this is possible to implement but only on Windows.

The implementation is reasonably tricky, alas, to do it properly it needs to be tightly integrated into the vfs cache and it will only work properly with remotes that implement change notify.

**Comment by silverkorn** [comment]
Was there work on this since then?
